### PR TITLE
update x/tools to fix panic in tests

### PR DESCRIPTION
**Pull request by alexandear**
This PR fixes panic in tests when running with Go 1.23.

Note, that the PR also changes to `go 1.18` in `go.mod` because `golang.org/x/tools` requires Go 1.22.

<details><summary>Details</summary>
<p>

```
❯ go version
go version go1.23.3 darwin/arm64
❯ go test
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x104a67794]

goroutine 137 [running]:
go/types.(*Checker).handleBailout(0x1400017c700, 0x140000c7c58)
	/opt/homebrew/Cellar/go/1.23.3/libexec/src/go/types/check.go:404 +0x9c
panic({0x104b86ae0?, 0x104d70640?})
	/opt/homebrew/Cellar/go/1.23.3/libexec/src/runtime/panic.go:785 +0x124
go/types.(*StdSizes).Sizeof(0x0, {0x104bd54a8, 0x104d73b80})
	/opt/homebrew/Cellar/go/1.23.3/libexec/src/go/types/sizes.go:229 +0x314
go/types.(*Config).sizeof(...)
	/opt/homebrew/Cellar/go/1.23.3/libexec/src/go/types/sizes.go:334
go/types.representableConst.func1({0x104bd54a8?, 0x104d73b80?})
	/opt/homebrew/Cellar/go/1.23.3/libexec/src/go/types/const.go:77 +0x90
go/types.representableConst({0x104bd6ad8, 0x104d67d48}, 0x1400017c700, 0x104d73b80, 0x140000c68b8)
	/opt/homebrew/Cellar/go/1.23.3/libexec/src/go/types/const.go:93 +0x134
go/types.(*Checker).representation(0x1400017c700, 0x140004b4100, 0x104d73b80)
	/opt/homebrew/Cellar/go/1.23.3/libexec/src/go/types/const.go:257 +0x68
go/types.(*Checker).implicitTypeAndValue(0x1400017c700, 0x140004b4100, {0x104bd54a8, 0x104d73b80})
	/opt/homebrew/Cellar/go/1.23.3/libexec/src/go/types/expr.go:377 +0x304
go/types.(*Checker).convertUntyped(0x1400017c700, 0x140004b4100, {0x104bd54a8, 0x104d73b80})
	/opt/homebrew/Cellar/go/1.23.3/libexec/src/go/types/const.go:290 +0x30
go/types.(*Checker).isValidIndex(0x1400017c700, 0x140004b4100, 0x34, {0x104ae87e9, 0x5}, 0x0)
	/opt/homebrew/Cellar/go/1.23.3/libexec/src/go/types/index.go:384 +0x64
go/types.(*Checker).index(0x1400017c700, {0x104bd5b80, 0x14000142260}, 0xffffffffffffffff)
	/opt/homebrew/Cellar/go/1.23.3/libexec/src/go/types/index.go:355 +0x9c
go/types.(*Checker).indexExpr(0x1400017c700, 0x140004b40c0, 0x1400015efa0)
	/opt/homebrew/Cellar/go/1.23.3/libexec/src/go/types/index.go:207 +0x6a0
go/types.(*Checker).exprInternal(0x1400017c700, 0x0, 0x140004b40c0, {0x104bd5f70, 0x1400010b260}, {0x0, 0x0})
	/opt/homebrew/Cellar/go/1.23.3/libexec/src/go/types/expr.go:1365 +0x208c
go/types.(*Checker).rawExpr(0x1400017c700, 0x0, 0x140004b40c0, {0x104bd5f70?, 0x1400010b260?}, {0x0?, 0x0?}, 0x0)
	/opt/homebrew/Cellar/go/1.23.3/libexec/src/go/types/expr.go:981 +0x120
go/types.(*Checker).expr(0x1400017c700, 0x0?, 0x140004b40c0, {0x104bd5f70?, 0x1400010b260?})
	/opt/homebrew/Cellar/go/1.23.3/libexec/src/go/types/expr.go:1549 +0x38
go/types.(*Checker).assignVar(0x1400017c700, {0x104bd5b50, 0x14000142220}, {0x104bd5f70, 0x1400010b260}, 0x0, {0x104aea027, 0xa})
	/opt/homebrew/Cellar/go/1.23.3/libexec/src/go/types/assignments.go:272 +0x104
go/types.(*Checker).assignVars(0x1400017c700, {0x140001123a0?, 0x1, 0x104bd54a8?}, {0x140001123b0, 0x104b90a20?, 0x140000c7298?})
	/opt/homebrew/Cellar/go/1.23.3/libexec/src/go/types/assignments.go:472 +0x2c0
go/types.(*Checker).stmt(0x1400017c700, 0x0, {0x104bd6210, 0x1400015c1c0})
	/opt/homebrew/Cellar/go/1.23.3/libexec/src/go/types/stmt.go:483 +0xd04
go/types.(*Checker).stmtList(0x1400017c700, 0x0, {0x140001423a0?, 0x1400017c700?, 0x11?})
	/opt/homebrew/Cellar/go/1.23.3/libexec/src/go/types/stmt.go:121 +0x88
go/types.(*Checker).funcBody(0x1400017c700, 0x104bd5b50?, {0x14000110120?, 0x104bd9098?}, 0x1400048d0c0, 0x1400010b350, {0x0?, 0x0?})
	/opt/homebrew/Cellar/go/1.23.3/libexec/src/go/types/stmt.go:41 +0x218
go/types.(*Checker).funcDecl.func1()
	/opt/homebrew/Cellar/go/1.23.3/libexec/src/go/types/decl.go:888 +0x44
go/types.(*Checker).processDelayed(0x1400017c700, 0x0)
	/opt/homebrew/Cellar/go/1.23.3/libexec/src/go/types/check.go:516 +0x12c
go/types.(*Checker).checkFiles(0x1400017c700, {0x140001140e8, 0x1, 0x1})
	/opt/homebrew/Cellar/go/1.23.3/libexec/src/go/types/check.go:462 +0x1b4
go/types.(*Checker).Files(0x1400015a000?, {0x140001140e8?, 0x14000148840?, 0x9?})
	/opt/homebrew/Cellar/go/1.23.3/libexec/src/go/types/check.go:422 +0x80
golang.org/x/tools/go/packages.(*loader).loadPackage(0x1400015a000, 0x140003c6560)
	/Users/Oleksandr_Redko/go/pkg/mod/golang.org/x/tools@v0.1.12/go/packages/packages.go:1001 +0x618
golang.org/x/tools/go/packages.(*loader).loadRecursive.func1()
	/Users/Oleksandr_Redko/go/pkg/mod/golang.org/x/tools@v0.1.12/go/packages/packages.go:838 +0x178
sync.(*Once).doSlow(0x0?, 0x0?)
	/opt/homebrew/Cellar/go/1.23.3/libexec/src/sync/once.go:76 +0xf8
sync.(*Once).Do(...)
	/opt/homebrew/Cellar/go/1.23.3/libexec/src/sync/once.go:67
golang.org/x/tools/go/packages.(*loader).loadRecursive(0x0?, 0x0?)
	/Users/Oleksandr_Redko/go/pkg/mod/golang.org/x/tools@v0.1.12/go/packages/packages.go:826 +0x48
golang.org/x/tools/go/packages.(*loader).loadRecursive.func1.1(0x0?)
	/Users/Oleksandr_Redko/go/pkg/mod/golang.org/x/tools@v0.1.12/go/packages/packages.go:833 +0x30
created by golang.org/x/tools/go/packages.(*loader).loadRecursive.func1 in goroutine 44
	/Users/Oleksandr_Redko/go/pkg/mod/golang.org/x/tools@v0.1.12/go/packages/packages.go:832 +0x84
exit status 2
FAIL	github.com/sivchari/containedctx	0.533s

```

</p>
</details>

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -12,22 +12,15 @@ jobs:
...
-        uses: actions/setup-go@v4
-        with:
-          go-version: '1.20'
-        id: go
-
       - name: Check out code into the Go module directory
-        uses: actions/checkout@v3
+        uses: actions/checkout@v4
 
-      - name: Restore cache
```
Comment (sivchari) [inline .github/workflows/ci.yml]: Why did you delete cache action ?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -12,22 +12,15 @@ jobs:
...
-        uses: actions/cache@v3
+      - name: Set up Go
+        uses: actions/setup-go@v5
         with:
-          path: ~/go/pkg/mod
-          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
-          restore-keys: |
-            ${{ runner.os }}-go-
+          go-version: '1.23'
```
Comment (sivchari) [inline .github/workflows/ci.yml]:
```suggestion
          go-version: '1.23.3'
```

File: go.mod [hunk 3]
```diff
@@ -1,18 +1,18 @@
 module github.com/sivchari/containedctx
 
-go 1.17
+go 1.18
```
Comment (sivchari) [inline go.mod]:
```suggestion
go 1.22.1

toolchain go1.23.3
```

**Review by sivchari — COMMENTED** [review]
I left nits, thanks !

File: .github/workflows/ci.yml (hunk 1 above)
Comment (alexandear) [reply .github/workflows/ci.yml]: It's redundant because `setup-go@v4` has enabled caching by default https://github.com/actions/setup-go#v4

File: .github/workflows/ci.yml (hunk 2 above)
Comment (alexandear) [reply .github/workflows/ci.yml]: I prefer `1.23` because it refers to the latest minor version. For example, when `1.23.4` becomes available, we don't need to upgrade the minor version here.

File: go.mod (hunk 3 above)
Comment (alexandear) [reply go.mod]: I can change to `1.22`, but in this way better is to use `1.22.3` and the latest `x/tools`

File: .github/workflows/ci.yml (hunk 1 above)
Comment (sivchari) [reply .github/workflows/ci.yml]: let's proceed in this direction, once. In fact, it's better no to use setup-go cache and use action/cache/restore and action/cache/save, but this problem is absolutely another thing. So let's handle it separately. Thanks.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (sivchari) [reply .github/workflows/ci.yml]: How about specifying stable ?

File: go.mod (hunk 3 above)
Comment (sivchari) [reply go.mod]: I prefer to use 1.22.1 since the golangci-lint uses go1.22.1 in themself. this linter might be almost used via golangci-lint.

**Review by sivchari — APPROVED** [review]
